QuadradiusR/scripts/embedded_server.gd

```diff
@@ -0,0 +1,87 @@
+extends Node
+
+var SERVER_PORT_FILE = '.server_port'
+var EXECUTABLE_PATH = './quadradiusr_server'
+
+var executable_path: String
+var available: bool
+var pid: int = -1
+var wait_left: float = 0
+var server_port: int = -1
+var on_server_ready_callbacks: Array = []
+
+func _init():
+	self.available = File.new().file_exists(EXECUTABLE_PATH)
+
+	if self.available:
+		print('Server executable is available')
+	else:
+		print('Server executable is not available')
+
+
+func _ready():
+	self.run_server()
+
+
+func run_server():
+	if not self.available:
+		push_error('Server executable is not available')
+
+	var file = File.new()
+
+	if file.file_exists(SERVER_PORT_FILE):
+		Directory.new().remove(SERVER_PORT_FILE)
+
+	var args = ['--embedded-mode', '--host', '0.0.0.0', '--port', '0']
+	print('Running server with arguments {args}'.format({'args': args}))
+	self.pid = OS.execute(EXECUTABLE_PATH, args, false, [])
+	self.wait_left = 0.05
+
+
+func stop_server():
+	if self.pid > 0:
+		OS.kill(self.pid)
+		self.pid = -1
+
+
+func is_available():
+	return self.available
+
+
+func get_pid():
+	return self.pid
+
+
+func get_port():
+	return self.server_port
+
+
+func on_server_ready(cb: FuncRef):
+	if self.server_port > 0:
+		cb.call_func()
+	self.on_server_ready_callbacks.append(cb)
+
+
+func _process(delta):
```

Comment:
This was not mentioned in our convention but I highly suggest moving this method higher up, close to ```_ready()``` since this handles our frame-by-frame logic
- [ ] move ```_process()`` function higher